tools/input-segments/src/InputSegments.ts

```diff
@@ -257,129 +271,214 @@ export class InputSegments extends TextfieldBase {
         };
 
         /**
-         * TODO: Include ARIA attributes for editable segments
-         * TODO: Use `@input`/`@beforeinput` events to handle data input/content cleanup
-         * TODO: Move `handleKeydown()` call to a cache so that it doesn't cycle on the binding in each render pass
-         * TODO: Rename `handleKeydown()` to match the new events used
+         * TODO: Include/review ARIA attributes for editable segments
+         * TODO: Move `handle` functions call to a cache so that it doesn't cycle on the binding in each render pass
          */
         return html`
             <div
                 role="spinbutton"
                 contenteditable=${ifDefined(isActive ? true : undefined)}
                 inputmode=${ifDefined(isActive ? 'numeric' : undefined)}
                 tabindex=${ifDefined(isActive ? '0' : undefined)}
-                class="editable-segment ${classMap(segmentClasses)}"
+                aria-hidden=${ifDefined(usePlaceholder ? 'true' : undefined)}
+                class=${classMap(segmentClasses)}
                 style=${styleMap(segmentStyles)}
                 data-testid=${segment.type}
                 @keydown=${(event: KeyboardEvent) => {
                     this.handleKeydown(segment, event);
                 }}
-            >
-                ${when(
-                    isPlaceholderVisible,
-                    () => html`
-                        <span aria-hidden="true" class="placeholder">
-                            ${segment.placeholder}
-                        </span>
-                    `,
-                    () => segment.formatted
-                )}
-            </div>
+                @beforeinput=${(event: InputEvent) => {
+                    this.handleBeforeInput(segment, event);
+                }}
+                @input=${(event: InputEvent) => {
+                    this.handleInputEvent(segment, event);
+                }}
+                .innerText=${this.renderSegmentText(segment)}
```

Comment:
We'll likely benefit from a comment on this one.

Reply:
Added